api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -264,4 +295,498 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() throws Exception {
 		//Then
 		assertTrue(concept.getSet());
 	}
+	
+	/**
+	 * @see ConceptServiceImpl#purgeConcept(Concept)
+	 * @Verifies should purge the concept if not being used by an obs
+	 */
+	@Test
+	public void purgeConcept_shouldPurgeTheConceptIfNotBeingUsedByAnObs() throws Exception {
+		int conceptId = 88;
+		conceptService.purgeConcept(conceptService.getConcept(conceptId));
+		assertNull(conceptService.getConcept(conceptId));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should fail if no reason is given
+	 */
+	@Test
+	public void retireConcept_shouldFailIfNoReasonIsGiven() throws Exception {
+		int conceptId = 3;
+		Concept concept = conceptService.getConcept(conceptId);
+		expectedException.expect(IllegalArgumentException.class);
+		conceptService.retireConcept(concept, "");
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should retire the given concept
+	 */
+	@Test
+	public void retireConcept_shouldRetireTheGivenConcept() throws Exception {
+		int drugId = 3;
```

Comment:
you dont need this local variable, only when using the same value more than once. like you did with the retireReason.